lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3888,6 +3888,10 @@ namespace {
         bodyParams =
             getNonSelfParamList(dc, decl, selfIdx, name.getArgumentNames(),
```

Comment:
Just a question, nonSelf here means it's a param list without self in it?

Reply:
Both `getNonSelfParamList` and `importFunctionParamsAndReturnType` call out to `importFunctionParameterList`. This one is just saying, if `selfIdx` is provided, ignore the parameter at that index because we'll import the self parameter later. That being said, I'm not sure what the case is where we'd have a self-parameter for a decl _other_ than a `CXXMethodDecl`. Something for me to look into.

Reply:
So to actually answer your question...

> Just a question, nonSelf here means it's a param list without self in it?

IIUC it's the opposite, this is intended to be used with a function that has a self parameter. 

Also, locally, I added `assert(!selfIdx);` and none of the tests crashed so I think this logic is wrong. I can submit a different PR to clean this up later.